### PR TITLE
reimplemented redis backed to use go-redis

### PR DESCRIPTION
**Pull request by sredxny**
In this branch was reimplemented the backend related to Redis, the main idea was to migrate the implementation from `redigo` to `go-redis`. This branch doesn't contains broken changes.
Is suggested to migrate TIB to use Go Modules as go-redis works better using go modules and they also suggest it as well.
The reason on reimplement this backend is to make it compatible with the dashboard so they can share the connection and not be forced to create 2 parallels connections for the same app when TIB runs as a feature of dashboard.

In order to work in harmony with dashboard as an internal feature is suggested to use the same go-redis version as dashboard.
Added support for sentinel, the implementation now looks similar to dashboard so they're compatible to share the connection

### CODE REVIEW

File: backends/redis.go [hunk 1]
```diff
@@ -31,79 +22,79 @@ type RedisConfig struct {
 	Hosts                 map[string]string
 	UseSSL                bool
 	SSLInsecureSkipVerify bool
+	Timeout               int
```
Comment (buger) [inline backends/redis.go]: See how it is implemented in the Gateway, Dash, or Pump, and do the same implementation. It has different arguments and also missing support for redis sentinel.

File: main.go [hunk 2]
```diff
@@ -33,12 +33,11 @@ var log = logger.Get()
 var mainLogger = log.WithField("prefix", "MAIN")
 var ProfileFilename *string
 
-// Get our bak end to use, new beack-ends must be registered here
+// Get our bak end to use, new back-ends must be registered here
```
Comment (matiasinsaurralde) [inline main.go]: `Get our backend`

**Comment by sredxny** [comment]
@buger updated the branch to support sentinel, cluster and single connection.

File: main.go [hunk 3]
```diff
@@ -33,12 +33,11 @@ var log = logger.Get()
...
 func initBackend(profileBackendConfiguration interface{}, identityBackendConfiguration interface{}) {
 
 	AuthConfigStore = &backends.InMemoryBackend{}
-	IdentityKeyStore = &backends.RedisBackend{KeyPrefix: "identity-cache."}
-
+	IdentityKeyStore = &backends.RedisBackend{KeyPrefix: "identity-cache"}
```
Comment (komalsukhani) [inline main.go]: Not sure if this is typo or intentional change

File: backends/redis.go [hunk 4]
```diff
@@ -1,109 +1,114 @@
...
 
-	redisLogger.Debug("Storage Engine already initialised...")
-	redisLogger.Debug("Redis handles: ", len(r.db.Handles))
+	r.db = r.newRedisClusterPool()
+	return true
+}
 
-	// Reset it just in case
-	r.db = redisClusterSingleton
+func (r *RedisBackend) cleanKey(keyName string) string {
```
Comment (komalsukhani) [inline backends/redis.go]: cleanKey function was added but is not used anywhere in the file

Comment (sredxny) [reply backends/redis.go]: done!

File: main.go (hunk 3 above)
Comment (sredxny) [reply main.go]: Please, can you elaborate more? I don't understand which typo

Comment (sredxny) [reply main.go]: The change on the KeyPrefix was to maintain an standard across all the other existing prefixes and have only text without dots, semi-colons, etc.

**Pull request merged**